cpp/velox/compute/VeloxInitializer.cc

```diff
@@ -112,18 +114,21 @@ void VeloxInitializer::Init(std::unordered_map<std::string, std::string>& conf)
   configurationValues.merge(S3Config);
 #endif
 
+  InitCache(conf);
+
   auto properties = std::make_shared<const velox::core::MemConfig>(configurationValues);
   auto hiveConnector =
       velox::connector::getConnectorFactory(velox::connector::hive::HiveConnectorFactory::kHiveConnectorName)
           ->newConnector(kHiveConnectorId, properties, ioExecutor_.get());
+  if (ioExecutor_) {
```

Comment:
Folly has an Executor implementation `QueuedImmediateExecutor` which supports running job from caller thread. Did you have a try on something like that?

Reply:
Or we must rely on async execution for works running on `ioExecutor_` and `ssdExecutor_` ?

Reply:
I'm not sure what you say, have you take a look for #1124 ? split preload logic in inside Velox, I think Gluten can use `QueuedImmediateExecutor ` to control it.

Reply:
@zhztheplayer yes, this patch fixes a bug in main branch

Reply:
@Yohahaha `folly::IOThreadPoolExecutor` is just an implementation of `folly::Executor`. Is this possible to use `folly::QueuedImmediateExecutor` to replace `folly::IOThreadPoolExecutor` to avoid Velox's async calls?